systems/notification/event-notify/pkg/server/event.go

```diff
@@ -52,684 +52,743 @@ func (es *EventToNotifyEventServer) EventNotification(ctx context.Context, e *ep
-			log.Errorf("unexpected item type for successful payment: %s", msg.ItemType)
-			break
+	case msgbus.PrepareRoute(es.orgName, evt.EventRoutingKey[evt.EventInvoiceGenerate]):
+		c := evt.EventToEventConfig[evt.EventInvoiceGenerate]
+		msg, err := epb.UnmarshalInvoiceGenerated(e.Msg, c.Name)
+		if err != nil {
+			return nil, err
 		}
+		handleEventInvoiceGenerate(es, msg, &c)
 
-		metadata := map[string]string{}
+	default:
+		log.Errorf("No handler routing key %s", e.RoutingKey)
+	}
 
-		err = json.Unmarshal(msg.Metadata, &metadata)
-		if err != nil {
-			log.Errorf("failed to Unmarshal payment metadata as map[string]string: %v", err)
-		}
+	return &epb.EventResponse{}, nil
+}
 
-		targetId, ok := metadata["targetId"]
-		if !ok {
-			log.Errorf("missing targetId metadata for successful package payment: %s", msg.ItemId)
-			break
-		}
+func handleEventOrgAdd(es *EventToNotifyEventServer, msg *epb.EventOrgCreate, c *evt.EventConfig) {
+	jmsg, err := json.Marshal(msg)
+	if err != nil {
+		log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)
+	}
+	es.processEvent(c, msg.Id, "", "", "", msg.Owner, jmsg, msg.Id)
+}
 
-		_ = es.ProcessEvent(&c, es.orgId, "", "", targetId, targetId, jmsg, msg.Id)
+func handleEventUserAdd(es *EventToNotifyEventServer, msg *epb.EventUserCreate, c *evt.EventConfig) {
+	jmsg, err := json.Marshal(msg)
+	if err != nil {
+		log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)
+	}
+	es.processEvent(c, es.orgId, "", "", "", msg.UserId, jmsg, msg.UserId)
+}
 
-	case msgbus.PrepareRoute(es.orgName, evt.EventRoutingKey[evt.EventPaymentFailed]):
-		c := evt.EventToEventConfig[evt.EventPaymentFailed]
-		msg, err := epb.UnmarshalPayment(e.Msg, c.Name)
-		if err != nil {
-			return nil, err
-		}
+func handleEventUserDeactivate(es *EventToNotifyEventServer, msg *epb.EventUserDeactivate, c *evt.EventConfig) {
+	jmsg, err := json.Marshal(msg)
+	if err != nil {
+		log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)
+	}
+	es.processEvent(c, es.orgId, "", "", "", msg.UserId, jmsg, msg.UserId)
+}
 
-		jmsg, err := json.Marshal(msg)
-		if err != nil {
-			log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)
-		}
+func handleEventUserDelete(es *EventToNotifyEventServer, msg *epb.EventUserDelete, c *evt.EventConfig) {
+	jmsg, err := json.Marshal(msg)
+	if err != nil {
+		log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)
+	}
+	es.processEvent(c, es.orgId, "", "", "", msg.UserId, jmsg, msg.UserId)
+}
 
-		if msg.ItemType != ukama.ItemTypePackage.String() {
-			log.Errorf("unexpected item type for successful payment: %s", msg.ItemType)
-			break
-		}
+func handleEventMemberCreate(es *EventToNotifyEventServer, msg *epb.AddMemberEventRequest, c *evt.EventConfig) {
+	jmsg, err := json.Marshal(msg)
+	if err != nil {
+		log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)
+	}
```

Comment:
same remark here ...

Also, since the marshalling is failling, meaning we are having an unvalid jmeg, why are we proceeding with the handling?  we should abort and return an error, right? 
`		log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)`
so I think in all these cases, we should maybe do more than logging: We should abort and return the error. 
Let me know what you think.

Reply:
addressed